From 4b92841ef27b56883fa4491a3d51db3eef68c481 Mon Sep 17 00:00:00 2001
From: Hyunchul Lee <hyc.lee@gmail.com>
Date: Sat, 10 Jul 2021 09:31:08 +0900
Subject: [PATCH] ksmbd: fix the running request count decrement
Git-commit: 4b92841ef27b56883fa4491a3d51db3eef68c481
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


decrement the count of running requests after
sending the last response for multi-response
requests.

Signed-off-by: Hyunchul Lee <hyc.lee@gmail.com>
Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/ksmbd/connection.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/fs/ksmbd/connection.c b/fs/ksmbd/connection.c
index 928e22e19def..6e51e08addee 100644
--- a/fs/ksmbd/connection.c
+++ b/fs/ksmbd/connection.c
@@ -120,7 +120,8 @@ int ksmbd_conn_try_dequeue_request(struct ksmbd_work *work)
 	    list_empty(&work->async_request_entry))
 		return 0;
 
-	atomic_dec(&conn->req_running);
+	if (!work->multiRsp)
+		atomic_dec(&conn->req_running);
 	spin_lock(&conn->request_lock);
 	if (!work->multiRsp) {
 		list_del_init(&work->request_entry);
-- 
2.33.0

